<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Js Errors</title>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-T3c6CoIi6uLrA9TneNEoa7RxnatzjcDSCmG1MXxSR1GAsXEV/Dwwykc2MPK8M2HN" crossorigin="anonymous">
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-C6RzsynM9kWDrMNeT87bh95OGNyZPhcTNXj1NW7RuBCsyN/o0jlpcV8Qyq46cDfL" crossorigin="anonymous"></script>
</head>
<body>
    
        <h2>Javascript try catch</h2>
    

        <p> Please input a number between 5 and 10:</p>
        <input type="text" name="number" id="demo">
        <button type="button" onclick="myfunc()">Test Input</button><br>
        <p id="p1">Enter the input to test</p>

    <script>
        function myfunc(){
            const message = document.getElementById('number')
            message.innerHTML = ""
            let x = document.getElementById('number').value
            try{
                if(x.trim() == "") throw "empty"
                if(isNaN(x)) throw "not a number"
                x = Number(x)
                if(x>10) throw("Number is too high")
                if(x<5) throw("Number is small")
            }
        catch(err){
            message.innerHTML = "Input is " + err
        }
        }
    </script>
</body>
</html>